recipes/newrelic/apm/php/debian-fpm.yml

```diff
@@ -21,25 +21,40 @@ keywords:
 processMatch:
   - php-fpm
 
-validationNrql: ""
+validationNrql: "SELECT count(*) from Transaction WHERE host like '{{.HOSTNAME}}%' facet entityGuid since 10 minutes ago"
 
+inputVars:
+  - name: "NR_PHP_APPLICATION"
+    prompt: "What is the name of your PHP application?"
+    default: "PHP Application"
+  - name: "NR_PHP_RESTART"
+    prompt: "Restart the PHP application after installation? (y/n)"
+    default: "y"
 
 install:
-
   version: "3"
   silent: true
 
+  vars:
+    TMP_INSTALL_DIR:
+      sh: mktemp -d /tmp/newrelic-php-agent.XXXXXX
+
   tasks:
     default:
       cmds:
         - task: verify_continue
         - task: assert_pre_req
+        - task: create_shared
+        - task: detect_services
         - task: add_gnupg2_curl_if_required
         - task: install_tarball
         - task: configure
         - task: start
```

Comment:
This task (and it's block [below](https://github.com/newrelic/open-install-library/blob/c600586e11ef2a2a12539a2cedb3756c266278d8/recipes/newrelic/apm/php/debian-fpm.yml#L261)) can be removed since the `restart services` task handles that.

Reply:
[Should](https://github.com/newrelic/open-install-library/blob/c600586e11ef2a2a12539a2cedb3756c266278d8/recipes/newrelic/apm/php/debian-fpm.yml#L201) navigate to `TMP_INSTALL_DIR` to install the tarball in.

Reply:
Since this section is expanding the tar logfile here, we don't need to do it [here](https://github.com/newrelic/open-install-library/pull/396/files#L299-L322), but the [logical](https://github.com/newrelic/open-install-library/pull/396/files#L299-L322) checks for the files should be moved up.  Additionally, since the first task already expanded the logfile, it can copy the expanded pertinent log file to a [global name](https://github.com/newrelic/open-install-library/pull/396/files#L39) that the second [task](https://github.com/newrelic/open-install-library/pull/396/files#L299-L322) can simply use.

Reply:
It might make sense to just do a `rm nrinstall*.log` before t[his line](https://github.com/newrelic/open-install-library/blob/c600586e11ef2a2a12539a2cedb3756c266278d8/recipes/newrelic/apm/php/debian-fpm.yml#L211) to ensure that [here](https://github.com/newrelic/open-install-library/blob/c600586e11ef2a2a12539a2cedb3756c266278d8/recipes/newrelic/apm/php/debian-fpm.yml#L230) we don't accidentally use an older file if one exists.

Reply:
[This](https://github.com/newrelic/open-install-library/blob/c600586e11ef2a2a12539a2cedb3756c266278d8/recipes/newrelic/apm/php/debian-fpm.yml#L218) was nicely done!! The modularization gives us so much more flexibility for adapting to different Linux flavors.

Reply:
Changes made / commentary:
* `start` has been removed in favor of `restart_services`
* There's no way to change where the log tarball is stored so these references to `/tmp` remain.
* Everything else should be using `{{.TMP_INSTALL_DIR}}`
* It does not look like global vars can reference previously defined global vars so I had to define `$NR_INSTALL_LOG` as an env var at the top of each place it is needed. We now create a symlink to a static name which at least avoids the need to run any command to get the log file name.
* The log file is now extracted in `{{.TMP_INSTALL_DIR}}` so I added in the removal of all agent install artifacts instead.
* The log file expansion logic has been deduped and now lives completely within `configure`. This has the added benefit of failing the install earlier if the tarball agent install fails.

Reply:
I thought doing the following:
```
cd /{{.TMP_INSTALL_DIR}}
wget http://download.newrelic.com/php_agent/archive/9.17.1.301/newrelic-php5-9.17.1.301-linux.tar.gz
tar xzf newrelic-php5-9.17.1.301-linux.tar.gz
sudo NR_INSTALL_USE_CP_NOT_LN=1 ./newrelic-php5-9.17.1.301-linux/newrelic-install install
```
installed it in the calling directory.
Does it always install in `/tmp` instead?

Reply:
The logfile is [always put](https://github.com/newrelic/newrelic-php-agent/blob/main/agent/newrelic-install.sh#L88-L92) in `/tmp` regardless of the working directory used to launch the installer.